dfpl/autoencoder.py

```diff
@@ -32,9 +31,13 @@ def define_ac_model(opts: options.Options, output_bias=None) -> Tuple[Model, Mod
     """
     input_size = opts.fpSize
     encoding_dim = opts.encFPSize
-    ac_optimizer = optimizers.Adam(
-        learning_rate=opts.aeLearningRate, decay=opts.aeLearningRateDecay
+    lr_schedule = optimizers.schedules.ExponentialDecay(
+        opts.aeLearningRate,
+        decay_steps=1000,
+        decay_rate=opts.aeLearningRateDecay,
+        staircase=True,
     )
+    ac_optimizer = optimizers.legacy.Adam(learning_rate=lr_schedule)
```

Comment:
I think it would be good to include necessary updates.

Reply:
what is a necessary update? since python 3.8 has reached its end of life, shall we update to 3.11 in this PR? this would require to also update chemprop since it runs on python versions < 3.9. But updating chemprop to the current one, requires a lot of code changes.